.github/workflows/test-environment.yml

```diff
@@ -229,6 +230,11 @@ jobs:
           aws s3 cp "${{ env.FLEET_API_DIR}}/kspm_eks.yaml" "${{ env.S3_BUCKET }}/kspm_eks.yaml"
           aws s3 cp "${{ env.FLEET_API_DIR}}/cspm-linux.sh" "${{ env.S3_BUCKET }}/cspm-linux.sh"
 
+      - name: Tag CNVM Instance
```

Comment:
I am not sure why this needs to be a separate step, can't the tags be added via terraform?

Reply:
No, CNVM is not deployed through Terraform. After provisioning the infrastructure using Terraform, the next steps involve installing CNVM integration and deploying an agent. However, it's important to note that the agent deployment is an asynchronous operation, and we don't wait until the stack is fully created. To save time and optimize the workflow, we proceed to the next actions and execute the CNVM tagging action separately.

Reply:
Understood

Then, perhaps we can add it to the script we have to deploy CloudFormation? This will also remove the requirement to add a step that waits for initialization.

Reply:
This is exactly the point: the `just deploy-cloudformation` command initiates the deployment of CloudFormation, but it doesn't wait for the stack creation to be completed. At this point in time, the `instance_id `doesn't exist yet because the EC2 instance associated with the agent is still being provisioned. In order to tag the EC2 instance associated with the agent, we need to obtain the instance_id. However, retrieving the `instance_id` is an asynchronous process, and we don't have visibility into when it will be available.

If we were to include the tagging command within the CloudFormation creation step, it would extend the overall workflow duration as we would have to wait for the `instance_id` before proceeding. This delay could potentially add several minutes to the overall runtime.

By providing a separate step for tagging, we can reduce the waiting time. This separation allows the tagging action to be performed "in parallel" (although it's not truly parallel) with other actions, thereby optimizing the overall workflow.

Regarding your mention of a step that waits for initialization, I'm not sure about its context. If you can provide more details, I can provide a more specific response

Reply:
I would even take it a step further and put the tagging step last. Only if the `cleanup-env` is set to false, as there is no need to tag an env that is supposed to be deleted, and if the delete step failed for some reason, the resources will be auto-deleted since they fail the tagging policy, wdyt?

Reply:
workflow is updated

Reply:
test flow is updated